apps/endatix-hub/components/survey.tsx

```diff
@@ -0,0 +1,32 @@
+'use client'
+
+import { Model } from 'survey-core'
+import { Survey } from 'survey-react-ui'
+import 'survey-core/defaultV2.css'
+import { startTransition } from "react";
+import { submitForm } from '@/app/(public)/share/[formId]/action.submit';
+  
+interface SurveyComponentProps {
+    definition: string;
+    formId: string;
+  }
+
+export default async function SurveyComponent({ definition, formId }: SurveyComponentProps) { 
+    const model = new Model(definition);
+    
+    const onFormComplete = async (sender: any, options: any) => {
+      const formData = JSON.stringify(sender.data, null, 3);
+      const metadata = {
+        notes: "",
+      };
+      const submissionData = {
+        jsonData: formData,
+        isComplete: true,
+        metadata: JSON.stringify(metadata, null, 3),
+      };
```

Comment:
This is way too common - we need to add an easy way to generate this boilerplate payload for sending a submission. I know this is what we have on the site. Just a remark, that we need to think of some utility code to help with this

Reply:
Sure. Also the submission object will acquire more properties in the future, once we start doing partial submissions and meta data.